dvc/repo/status.py

```diff
@@ -113,4 +114,15 @@ def status(
             remote=remote,
             all_tags=all_tags,
         )
+
+    ignored = list(
+        compress(
+            ["all_branches", "all_tags", "jobs", "remote", "cloud"],
+            [all_branches, all_tags, jobs, remote, cloud],
+        )
+    )
+    if ignored:
+        msg = "the following options are meaningless for local status: {}"
+        logger.warning(msg.format(", ".join(ignored)))
+
```

Comment:
How about we actually raise an exception here or in dvc/command/status (-c is required for those flags) instead of adding this warning? The warning looks like a UI pollution to me.

Reply:
Raising an exception is a bit backward incompatible.

Reply:
@Suor I think that is acceptable since the previous behaviour was incorrect anyway.